### PR TITLE
Editing of parking permit layout

### PR DESCRIPTION
**Pull request by amanpdyadav**
* Navigate user to valid permit component
* Add margin below receipt download button
* Update permit component
* Set default selected address if valid permit exist
* Allow max 2 permit per user
* Show secondary address only if it exists
* Disable address selection if user have valid permit
* Separate valid permits and drafts
* Condition for showing validPermit list
* Add stepper for validPermit component
* Add PermitStatus types
* Add translations from validPermit components
* Add validPermit component
* Add translations for addressLabel component
* Add address label component
* Add parkingZone node to graphql requests
* Add primary filed to profile query address object
* Fix userProfile address type
* Fix lang warning from console
* Fetch zone price from the backend
* Add street number to address formatter

Screenshot 2021-09-24 at 11 08 21
Screenshot 2021-09-24 at 11 11 27
Screenshot 2021-09-24 at 11 12 08

### CODE REVIEW

File: src/redux/reducers/permitCart.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import {
 import { PermitCartState, ProcessingStatus } from '../types';
 
 const initialState: PermitCartState = {
-  currentStep: 1,
+  currentStep: 0,
```
Comment (mingfeng) [inline src/redux/reducers/permitCart.ts:20]: Maybe use `STEPPER.VALID_PERMITS`

File: src/common/vehicleSelector/VehicleSelector.tsx [hunk 2]
```diff
@@ -40,6 +41,10 @@ const VehicleSelector = ({
       message: t(`${T_PATH}.notification.info.message`),
     },
   ];
+  const allPermits = () =>
```
Comment (mingfeng) [inline src/common/vehicleSelector/VehicleSelector.tsx:44]: Maybe we could have a better name for it, like `numRegistrations`

Comment (amanpdyadav) [reply src/common/vehicleSelector/VehicleSelector.tsx]: Done renamed it to `numOfUserCars`

File: src/redux/reducers/permitCart.ts (hunk 1 above)
Comment (amanpdyadav) [reply src/redux/reducers/permitCart.ts]: Done.

File: src/common/permit/Permit.tsx [hunk 3]
```diff
@@ -48,43 +65,72 @@ const Permit = ({
...
+        <AddressLabel address={address} />
+        <ParkingZonesMap userAddress={address} zoom={13} />
+        {showActionsButtons && showChangeAddressButtons && (
+          <div className="permit-action-btns">
+            <Button
+              className="permit-actions-buttons"
+              variant="supplementary"
+              style={{ margin: 'var(--spacing-xs) 0' }}
+              iconLeft={<IconAngleRight />}>
+              {t(`Vaihda osoitetta`)}
```
Comment (mingfeng) [inline src/common/permit/Permit.tsx:103]: I think it can be a more proper translation key?

File: src/common/permit/Permit.tsx [hunk 4]
```diff
@@ -48,43 +65,72 @@ const Permit = ({
...
+      <div className="permit-card">
+        {permits.map((permit, index) => (
+          <Card
+            key={uuidv4()}
+            style={{ marginTop: index > 0 ? 'var(--spacing-xs)' : '0' }}>
+            {getPermit(permit)}
+            {showActionsButtons && (
+              <div className="permit-action-btns">
+                <Button variant="supplementary" iconLeft={<IconAngleRight />}>
+                  {t(`Muokkaa ajoneuvon tietoja`)}
```
Comment (mingfeng) [inline src/common/permit/Permit.tsx:117]: ditto

File: src/common/permit/Permit.tsx [hunk 5]
```diff
@@ -48,43 +65,72 @@ const Permit = ({
...
+                </Button>
+                {permits.length > 1 && index > 0 && (
+                  <Button
+                    variant="supplementary"
+                    iconLeft={<IconAngleRight />}
+                    onClick={() =>
+                      dispatch(deletePermit(user as UserProfile, permit.id))
+                    }>
+                    {t(`Poista toinen ajoneuvo`)}
```
Comment (mingfeng) [inline src/common/permit/Permit.tsx:126]: ditto

**Review by mingfeng — COMMENTED** [review]
Some translation keys need to be updated, otherwise LGTM

File: src/common/permit/Permit.tsx (hunk 3 above)
Comment (amanpdyadav) [reply src/common/permit/Permit.tsx]: Added.

File: src/common/permit/Permit.tsx (hunk 4 above)
Comment (amanpdyadav) [reply src/common/permit/Permit.tsx]: Added.

File: src/common/permit/Permit.tsx (hunk 5 above)
Comment (amanpdyadav) [reply src/common/permit/Permit.tsx]: Added.

**Review by mingfeng — APPROVED** [review]
LGTM 👍

**Pull request merged**